tests/test_dataset.py

```diff
@@ -0,0 +1,426 @@
+# Copyright (C) 2017-2022  Cleanlab Inc.
+# This file is part of cleanlab.
+#
+# cleanlab is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# cleanlab is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with cleanlab.  If not, see <https://www.gnu.org/licenses/>.
+
+import requests
+import pytest
+import io
+import numpy as np
+from cleanlab import dataset
+
```

Comment:
May want test for edge-case with only 2 classes as well?  I could envision potential bugs / formatting issues in that edge-case

Reply:
yup included in current tests (imdb is 2 classes)